core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -530,6 +544,16 @@ private void generateManifest(FileSystem runnerZipFs, final String classPath, Pa
             }
         }
         attributes.put(Attributes.Name.MAIN_CLASS, config.mainClass);
+        if (config.manifest.implementationEntries && !attributes.containsKey(Attributes.Name.IMPLEMENTATION_TITLE)) {
+            String name = applicationInfo.getName().equals(UNSET_VALUE) ? appArtifact.getArtifactId()
```

Comment:
It doesn't really matter in this case but it's good practice to do it the other way around:
```
UNSET_VALUE.equals(applicationInfo.getName())
```
It automatically protects you from null values.